src/NumSharp.Core/Utilities/Maths/Operator.Bitwise.cs

```diff
@@ -0,0 +1,188 @@
+        [MethodImpl(MethodImplOptions.AggressiveInlining)] public static uint BitwiseAnd(uint lhs, byte rhs) => lhs & rhs;
+        [MethodImpl(MethodImplOptions.AggressiveInlining)] public static long BitwiseAnd(uint lhs, short rhs) => lhs & rhs;
+        [MethodImpl(MethodImplOptions.AggressiveInlining)] public static uint BitwiseAnd(uint lhs, ushort rhs) => lhs & rhs;
+        [MethodImpl(MethodImplOptions.AggressiveInlining)] public static long BitwiseAnd(uint lhs, int rhs) => (int)lhs & rhs;
+        [MethodImpl(MethodImplOptions.AggressiveInlining)] public static uint BitwiseAnd(uint lhs, uint rhs) => lhs & rhs;
+        [MethodImpl(MethodImplOptions.AggressiveInlining)] public static long BitwiseAnd(uint lhs, long rhs) => lhs & rhs;
+        [MethodImpl(MethodImplOptions.AggressiveInlining)] public static ulong BitwiseAnd(uint lhs, ulong rhs) => lhs & rhs;
+
+        // Int64
+        // print((np.array([1], dtype=np.int64) & np.array([True])).dtype)                # -> int64 : long
+        // print((np.array([1], dtype=np.int64) & np.array([1], dtype=np.int8)).dtype)    # -> int64 : long
+        // print((np.array([1], dtype=np.int64) & np.array([1], dtype=np.uint8)).dtype)   # -> int64 : long
+        // print((np.array([1], dtype=np.int64) & np.array([1], dtype=np.int16)).dtype)   # -> int64 : long
+        // print((np.array([1], dtype=np.int64) & np.array([1], dtype=np.uint16)).dtype)  # -> int64 : long
+        // print((np.array([1], dtype=np.int64) & np.array([1], dtype=np.int32)).dtype)   # -> int64 : long
+        // print((np.array([1], dtype=np.int64) & np.array([1], dtype=np.uint32)).dtype)  # -> int64 : long
+        // print((np.array([1], dtype=np.int64) & np.array([1], dtype=np.int64)).dtype)   # -> int64 : long
+        [MethodImpl(MethodImplOptions.AggressiveInlining)] public static long BitwiseAnd(long lhs, bool rhs) => lhs & Convert.ToInt64(rhs);
+        [MethodImpl(MethodImplOptions.AggressiveInlining)] public static long BitwiseAnd(long lhs, sbyte rhs) => lhs & rhs;
+        [MethodImpl(MethodImplOptions.AggressiveInlining)] public static long BitwiseAnd(long lhs, byte rhs) => lhs & rhs;
+        [MethodImpl(MethodImplOptions.AggressiveInlining)] public static long BitwiseAnd(long lhs, short rhs) => lhs & rhs;
+        [MethodImpl(MethodImplOptions.AggressiveInlining)] public static long BitwiseAnd(long lhs, ushort rhs) => lhs & rhs;
+        [MethodImpl(MethodImplOptions.AggressiveInlining)] public static long BitwiseAnd(long lhs, int rhs) => lhs & rhs;
+        [MethodImpl(MethodImplOptions.AggressiveInlining)] public static long BitwiseAnd(long lhs, uint rhs) => lhs & rhs;
+        [MethodImpl(MethodImplOptions.AggressiveInlining)] public static long BitwiseAnd(long lhs, long rhs) => lhs & rhs;
+        [MethodImpl(MethodImplOptions.AggressiveInlining)] public static long BitwiseAnd(long lhs, ulong rhs) => throw new NotSupportedException("The operator is 'bitwise_and' not supported for the input types.");
```

Comment:
I've created my own implementation of [Converts](https://github.com/SciSharp/NumSharp/blob/master/src/NumSharp.Core/Utilities/Converts.Native.cs) that ensures it is always inlined to save some extra ops inside closely-tight loops. Converts also has additional features in it, have a look.